docs/cargo_features.md

```diff
@@ -39,6 +39,13 @@
 |flac|FLAC audio format support. It's included in bevy_audio feature.|
 |mp3|MP3 audio format support.|
 |wav|WAV audio format support.|
+|symphonia-aac|AAC audio format support by Symphonia.|
+|symphonia-all|AAC, FLAC, MP4, MP3, Vorbis, WAV support by Symphonia.|
+|symphonia-flac|FLAC audio format support by Symphonia.|
```

Comment:
It would be nice to have a short explanation of how this is different than the `flac` feature (same for `mp3` and `wav`)

Reply:
Do I put the explanation in this cargo_features file as well? Besides, AFAIK they are just different libraries. Both can be used to process audio files. A benefit I could think of using Symphonia is easily change audio formats used by games since it can probe the media source and decode accordingly instead of changing bunch of code when you need to change audio file from wav = hound to mp3 = minimp3.

Reply:
Bevy completely abstracts over the file loaded, so changing from wav to Flac would just mean changing the feature and the file extension.

I would like to see in this file why someone would enable a feature, and I'm not sure "support by Symphonia" is enough of a difference with the `flac` feature to help users decide which they need.

Reply:
Internally, rodio parses audio files using the following [libraries](https://github.com/RustAudio/rodio/blob/70c80462e64d46235c7b7dd836e735ef89e29ddd/Cargo.toml#L15-L18):

| Format| Library |
| --- | --- |
| flac | claxon |
| vorbis | lewton |
| wav | hound |
| mp3 | minimp3 |

Symphonia support parsing all of the mentioned file formats and more, however it is much "heavier" due to having support of more file formats. Therefore symphonia is not enabled by default.

Symphonia is usually enabled if:

1. You have audio files not supported by any of the four audio files mentioned. (I don't think there's even a library for parsing `aac` files that is widely used and tested)
2. You support a multitude of audio file formats and would not like to compile five different crates that essentially do the same thing (well, more than five, since symphonia have its own subcrates)

In this scenario, symphonia is just an implementation detail. Perhaps we can indicate that when enabling symphonia, they should disable other features to avoid redundancies. (also when you enable both `mp3` and `symphonia-mp3` features, the `mp3` will actually be disable and symphonia will take priority. see [here](https://github.com/RustAudio/rodio/blob/0356d810c9e7309c12f6e4c20f0add64420ecf57/src/decoder/mod.rs#L47-L54))

As such, enabling all the aformentioned features plus `symphonia-all` will simply render the above four features useless, while still compiling the corresponding libraries (i think. haven't delved deep into how rust compile dependencies).

Maybe the best thing to do is to link symphonia to its docs.rs page, and mention this comment (or put it) on `symphonia-all`'s description.

Reply:
I'd be very happy to add a modified version of this comment to the feature list.

Reply:
Just a few more info to chip in after researching a bit. 

On the codec `mp3`, the `rodio` is most likely also switching to `symphonia` due to cross compilations, potentially unsafe code and perhaps ios rejection as discussed [here](https://github.com/RustAudio/rodio/issues/409). 

And also over [here](https://github.com/RustAudio/rodio/issues/176), it seems `symphonia` has more momentums compared to `lewton` for `vorbis` and therefore, could reiterate more often and provide more features/fixes perhaps?